### PR TITLE
pythonPackages.m3u8: init at 0.5.2

### PR DESCRIPTION
**Pull request by Scriptkiddi**
###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Review by jonringer — COMMENTED** [review]
There's no tests located in the pypi package, do you mind checking out the code from github:

```
running build_ext

----------------------------------------------------------------------
Ran 0 tests in 0.000s

OK
Finished executing setuptoolsCheckPhase
/nix/store/93asqnc8d8q1bpw32dfc198xfn9wxx2s-python3.7-m3u8-0.5.2
```

**Comment by Scriptkiddi** [comment]
the python test are done running runtests and it sets up a http server, is that what we want in a python test?

**Comment by jonringer** [comment]
There's some tests such as https://github.com/globocom/m3u8/blob/master/tests/test_parser.py which seem to be unit tests. 

It doesn't have to be full integration tests, but tests are useful to ensure that the application still works in most use cases.

File: pkgs/development/python-modules/m3u8/default.nix [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1kbfrdw1dbrjdqxj23r4b5hwc7b5dmlpprs3ryia4wnbsfygx54w";
+  };
+
+  propagatedBuildInputs = [ requests iso8601 ];
+
+  meta = with lib; {
+    homepage = "https://github.com/globocom/m3u8";
+    description = "Python m3u8 parser.";
```
Comment (costrouc) [inline pkgs/development/python-modules/m3u8/default.nix]: Description should not end with period

**Comment by Scriptkiddi** [comment]
@jonringer added the tests

File: pkgs/development/python-modules/m3u8/default.nix [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    rev = version;
+    sha256 = "0p6wmwv1nfa5pyakq5d55w9v142z5ja3db3s3qr44kx895d9lhng";
+  };
+
+  checkInputs = [ bottle pytest pytestcov ];
+
+  checkPhase = ''
+    ${python.interpreter} -m pytest tests/test_parser.py
+    ${python.interpreter} -m pytest tests/test_model.py
+    ${python.interpreter} -m pytest tests/test_variant_m3u8.py
```
Comment (jonringer) [inline pkgs/development/python-modules/m3u8/default.nix]: this can be simply:
```
  pytest tests/test_{parser,model,variant_m3u8}.py
```

Comment (Scriptkiddi) [reply pkgs/development/python-modules/m3u8/default.nix]: done

Comment (jonringer) [reply pkgs/development/python-modules/m3u8/default.nix]: pytest can accept multiple paths, and {...} is bash expansion

File: pkgs/development/python-modules/m3u8/default.nix [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    rev = version;
+    sha256 = "0p6wmwv1nfa5pyakq5d55w9v142z5ja3db3s3qr44kx895d9lhng";
+  };
+
+  checkInputs = [ bottle pytest pytestcov ];
+
+  checkPhase = ''
+    pytest tests/test_parser.py
+    pytest tests/test_model.py
+    pytest tests/test_variant_m3u8.py
```
Comment (jonringer) [inline pkgs/development/python-modules/m3u8/default.nix]: you can replace those 3 lines, with below
```suggestion
     pytest tests/test_{parser,model,variant_m3u8}.py
```

File: pkgs/development/python-modules/m3u8/default.nix [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+    repo = pname;
+    rev = version;
+    sha256 = "0p6wmwv1nfa5pyakq5d55w9v142z5ja3db3s3qr44kx895d9lhng";
+  };
+
+  checkInputs = [ bottle pytest pytestcov ];
+
+  checkPhase = ''
+    pytest tests/test_parser.py
+    pytest tests/test_model.py
```
Comment (jonringer) [inline pkgs/development/python-modules/m3u8/default.nix]:
```suggestion
```

File: pkgs/development/python-modules/m3u8/default.nix [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    owner = "globocom";
+    repo = pname;
+    rev = version;
+    sha256 = "0p6wmwv1nfa5pyakq5d55w9v142z5ja3db3s3qr44kx895d9lhng";
+  };
+
+  checkInputs = [ bottle pytest pytestcov ];
+
+  checkPhase = ''
+    pytest tests/test_parser.py
```
Comment (jonringer) [inline pkgs/development/python-modules/m3u8/default.nix]:
```suggestion
```

File: pkgs/development/python-modules/m3u8/default.nix [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    pytest tests/test_model.py
+     pytest tests/test_{parser,model,variant_m3u8}.py
```
Comment (jonringer) [inline pkgs/development/python-modules/m3u8/default.nix]:
```suggestion
    pytest tests/test_{parser,model,variant_m3u8}.py
```

**Review by jonringer — COMMENTED** [review]
```
pytest tests/test_{parser,model,variant_m3u8}.py
```
runs the same tests as 
```
pytest tests/test_parser.py
pytest tests/test_model.py
 pytest tests/test_variant_m3u8.py
```
but it collects them all into a single run (which is easier to digest)

**Review by jonringer — APPROVED** [review]
diff LGTM
commits LGTM
has tests :+1: 

```
[5 built, 54 copied (21.9 MiB), 4.1 MiB DL]
https://github.com/NixOS/nixpkgs/pull/74110
3 package were built:
python27Packages.m3u8 python37Packages.m3u8 python38Packages.m3u8
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build python27Packages.m3u8 python37Packages.m3u8 python38Packages.m3u8

**Pull request merged**